### ISSUE TITLE
Improve file tree accessibility

### ISSUE THREAD

**Issue by umpox** [sourcegraph/sourcegraph#19167]
### Code:
Example: https://sourcegraph.com/github.com/sourcegraph/sourcegraph/-/blob/client/web/src/tree/TreeLayer.tsx#L290
Example: https://sourcegraph.com/github.com/sourcegraph/sourcegraph/-/blob/client/web/src/tree/SingleChildTreeLayer.tsx#L128

Screenshot:
![image](https://user-images.githubusercontent.com/9516420/111305484-eacff380-864e-11eb-8707-1a9ba256f8ca.png)

Problems:
1. We manage keyboard navigation and focus logic ourselves, it is not possible for the user to actually move their focus to specific elements in this list
2. We prefetch file data `onMouseOver` on the list, but we only support the same functionality for keyboard focusing in `componentDidUpdate`. We should be able to use `onFocus` here.

**Comment by felixfbecker** [comment]
I think this is a duplicate of https://github.com/sourcegraph/sourcegraph/issues/12916 (minus the `onMouseOver` behavior, nice catch on that)

**Comment by umpox** [comment]
@felixfbecker Thanks! Will close this one.
